torch/_inductor/output_code.py

```diff
@@ -68,32 +67,30 @@
     from .triton_bundler import TritonKernelArtifacts
 
 
-class OutputCode(Protocol):
+@dataclasses.dataclass
+class OutputCode:
+    # TODO: Remove underscores here
+
+    # None if the output is not remote cacheable
+    _fx_graph_cache_key: Optional[str] = dataclasses.field(default=None, init=False)
```

Comment:
Do we think this should be a method instead? And then have each implementation decide how to implement the method?

Reply:
I am still trying to understand how this field is populated, but my current impression is that we bong these attributes in no matter what concrete OutputCode subclass we have, so no, having it be a property under this design is probably fine.